phd-tests/framework/src/test_vm/mod.rs

```diff
@@ -142,6 +142,51 @@ enum VmState {
     Ensured { serial: SerialConsole },
 }
 
+/// Both the output and status of a command.
+pub struct ShellOutput {
+    status: u16,
+    output: String,
+}
+
+impl ShellOutput {
+    /// Consume this [`ShellOutput`], returning the command's output as text
+    /// if the command completed successfully, or an error if it did not.
+    pub fn expect_ok(self) -> Result<String> {
```

Comment:
it's a bit weird to me that `expect_ok()` returns a `Result` instead of panicking --- i would _generally_ expect (haha) a function named `expect_foo` to panic if the expectation is violated. Perhaps these would make more sense as just `.ok()`/`.err()`? I dunno. Maybe this is clearer to others...

Reply:
hum yeah that seems pretty reasonable. this naming _probably_ made a little more sense in the first draft where i was hoping this would configure a future to _expect the result is ok_ rather than this `goofy-Result-to-Result` transformation. i'll see if i can get something nice with `IntoFuture` and if not i'll `s/expect_/g`